### ISSUE TITLE
Workspace - The invite message can't fit into the textbox

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#37406]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
**Found when validating PR :** https://github.com/Expensify/App/pull/29991

**Version Number:** 1.4.44.0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Issue reported by:** Applause-Internal Team

## Action Performed:
1. Log in with a Gmail account
2. Create a new workspace
3. Tap on the "Invite members" button
4. Select any suggested member
5. Tap on the Invite button

## Expected Result:
The invite message should fit into the textbox.

## Actual Result:
The invite message can't fit into the textbox.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [x] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

![Bug6395091_1709098760283!capture](https://github.com/Expensify/App/assets/115492554/c6e8d26e-8a51-4c18-b70a-4c91bc0413dd)

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by izarutskaya** [comment]
@mallenexpensify I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors.

**Comment by mallenexpensify** [comment]
_Think_ it can be external. @paultsimura , let me know if you disagree. 
Checking with Wave 6, collect submitter for project
https://expensify.slack.com/archives/C02MW39LT9N/p1709337922878799

**Comment by paultsimura** [comment]
> _Think_ it can be external.

Yes, it seems like a regular FE task 👍

**Comment by mallenexpensify** [comment]
Checking in Wave 8, think it's better https://expensify.slack.com/archives/C036QM0SLJK/p1709340832943519

**Comment by godofoutcasts94** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Workspace - The invite message can't fit into the textbox
### What is the root cause of that problem?
Here, 
https://github.com/Expensify/App/blob/af68510ad375bc99679908ce26dafead3a8b3dc1/src/pages/workspace/WorkspaceInviteMessagePage.tsx#L178
we are passing `autoGrowHeightMultilineInput`, which has a max height of `maxHeight: 115`which is basically clipping the text and because of this the invite message does not fit inside the textbox.
### What changes do you think we should make in order to solve the problem?

To solve this issue,  I can see that the app behaves by having a scrollbar in the textbox on all the other platforms i.e. 
-  Android: Native
-  iOS: Native
- MacOS: Chrome / Safari
- MacOS: Desktop
So, the best solution is that we can add `overflow: scroll` here :

https://github.com/Expensify/App/blob/a50923478ce163873c905e3566ccdc22f5284ce1/src/styles/index.ts#L3124-L3126

### What alternative solutions did you explore? (Optional)
Other, than that we can do one more thing is we can make the textbox, height dynamic i.e. it changes according to the text input.

**Comment by paultsimura** [comment]
After taking a closer look – I'm not sure it's something we need to fix.
The issue is that the welcome message is taller than the defined max height, and the message can be of any height, so there's no sense to just increase the max height to some other fixed value.

However, the field is scrollable. Do we want to change this behavior? And if yes – what would we expect to be the outcome?

https://github.com/Expensify/App/assets/12595293/27e855e5-1fab-4d9b-96a7-a36717480446



cc @mallenexpensify @Expensify/design

**Comment by godofoutcasts94** [comment]
> After taking a closer look – I'm not sure it's something we need to fix. The issue is that the welcome message is taller than the defined max height, and the message can be of any height, so there's no sense to just increase the max height to some other fixed value.
> 
> However, the field is scrollable. Do we want to change this behavior? And if yes – what would be the expected outcome?
> 
>  Simulator.Screen.Recording.-.iPhone.15.Pro.-.2024-03-02.at.12.43.51.mp4 
> cc @mallenexpensify @Expensify/design

@paultsimura, As stated above, I can understand that and see that. Also, my solution revolves more around making it scrollable as I have also stated that this kind of behaviour is experienced in the app on other platforms. Please checkout my [proposal](https://github.com/Expensify/App/issues/37406#issuecomment-1974702241) if possible for you

**Comment by mallenexpensify** [comment]
@shawnborton looping in design for 👀. 
Also, I wasn't able to reproduce on iOS Safari

![IMG_1208](https://github.com/Expensify/App/assets/22508304/c9403fa1-9b9c-44bc-b95a-19b212c8a5e6)

One possible discussion point, if we end up having too much text and the user needs to scroll, when the user first sees the message, do we want to show them the top of the text or the bottom, assuming they're able to scroll down.

**Comment by paultsimura** [comment]
> Also, I wasn't able to reproduce on iOS Safari

Just duplicate your invite message to make it 2x as long

**Comment by godofoutcasts94** [comment]
> > The issue is that the welcome message is taller than the defined max height, and the message can be of any height, so there's no sense to just increase the max height to some other fixed value.
> 
> @paultsimura I think the problem here is that the "default invite message" is clipped when it's shown to the user, it's supposed to always be showing in full when the user opens that page (if not the user will have to scroll to read the full text). The current `max height` was set based on that assumption too.
> 
> > Just duplicate your invite message to make it 2x as long
> 
> If the user edits the invite message and it becomes longer than the text box, it's not a problem because at the time the user already reads the full text of the "default invite message"

But I think making it scrollable is a better alternative as in all the other platforms we are just making the invite message as scrollable. After that we can add as much text in the invite message as possible and it won't effect anything and also it would maintain consistency throughout the app on all the platforms.

**Comment by paultsimura** [comment]
> I think making it scrollable is a better alternative as in all the other platforms we are just making the invite message as scrollable

From what I see – we have this message scrollable on all platforms. Could you please share the recordings on what platforms the message is not scrollable?

**Comment by paultsimura** [comment]
> If the user edits the invite message and it becomes longer than the text box, it's not a problem because at the time the user already reads the full text of the "default invite message"

If I modify the message, leave the page, and go back to it – the new custom message will be "default" for all future invitations. So this is not necessarily related to the default invitation message set by Expensify.


https://github.com/Expensify/App/assets/12595293/a01cdbc8-b772-40bd-82fa-2f180802e36d

**Comment by godofoutcasts94** [comment]
> > I think making it scrollable is a better alternative as in all the other platforms we are just making the invite message as scrollable
> 
> From what I see – we have this message scrollable on all platforms. Could you please share the recordings on what platforms the message is not scrollable?

Yes yes I know that and that's what I am saying that we should make the `mweb Chrome` and `mweb IOS` scrollable.

**Comment by paultsimura** [comment]
> Yes yes I know that and that's what I am saying that we should make the `mweb Chrome : Android` and `mweb Safari : IOS` scrollable for invite message textbox

@godofoutcasts94 please try to keep the communication productive. From what I see, all the platforms (including both mWeb) already have this field scrollable. If you have a different experience - please provide the screen recording.

**Comment by godofoutcasts94** [comment]
> > Yes yes I know that and that's what I am saying that we should make the `mweb Chrome : Android` and `mweb Safari : IOS` scrollable for invite message textbox
> 
> @godofoutcasts94 please try to keep the communication productive. From what I see, all the platforms (including both mWeb) already have this field scrollable. If you have a different experience - please provide the screen recording.

oops my bad! I am sorry. I had some misunderstanding. @paultsimura I am extremely sorry.

**Comment by paultsimura** [comment]
@shawnborton just so you don't have to scroll through all the messages.

**TD;DR:**
The issue is that the welcome message is taller than the defined max height, and the message can be of any height, so there's no sense to just increase the max height to some other fixed value.

However, the field is scrollable. Do we want to change this behavior? And if yes – what would be the expected outcome?

https://github.com/Expensify/App/assets/12595293/27e855e5-1fab-4d9b-96a7-a36717480446


https://github.com/Expensify/App/issues/37406#issuecomment-1974774450
https://github.com/Expensify/App/issues/37406#issuecomment-1976413695

**Comment by dannymcclain** [comment]
I'm down to wait for the other designers to chime in, but I think this is expected behavior and isn't something we need to do anything about. Looks to be working like a normal `textarea` input, which I think is a good and fine thing.

>One possible discussion point, if we end up having too much text and the user needs to scroll, when the user first sees the message, do we want to show them the top of the text or the bottom, assuming they're able to scroll down.

@mallenexpensify I think it kinda depends on whether the field is focused or not. If the field is focused, we should continue to default to the end/bottom, because that's where we place your cursor in the input (I assume). If the field is _not_ focused, then I could see an argument for defaulting to the top of the message, but honestly think either is fine.

**Comment by shawnborton** [comment]
Yeah I agree with Danny here - this seems to be standard textarea behavior. 

Did we recently change the max height or something?

**Comment by paultsimura** [comment]
> Did we recently change the max height or something?

Nope. I think the issue reporter just had a relatively small screen and a relatively long workspace name, so the default message overflowed the pre-defined textarea height.

So would you suggest to close this issue? @shawnborton @dannymcclain

**Comment by shawnborton** [comment]
I would be good with that, yes.

**Comment by mallenexpensify** [comment]
Thanks all, closing
